app/models/user.rb

```diff
@@ -94,7 +94,7 @@ def self.from_token(token:)
     auth = auth.stringify_keys
     args = auth.slice(*ATTRIBUTES)
 
-    args['id'] ||= auth['sub']
+    args['id'] ||= auth['sub'].sub('student:', '')
```

Comment:
this looks a little hacky, why would an id contain the prefix `student:`?

Reply:
it does feel a little hacky but I found student IDs are always prefixed (see [here](https://github.com/RaspberryPiFoundation/profile/blob/main/app/lib/subject.js#L21-L26)) which was breaking everything on our end and thought this was the best way to solve that